### ISSUE TITLE
Add transactional enqueuing

### ISSUE THREAD

**Issue by nathankawalec** [craigpastro/pgmq-go#53]
It would be very beneficial to be able to queue messages in a transaction, similar to what riverqueue offers (https://pkg.go.dev/github.com/riverqueue/river#Client.InsertTx), this would completely mitigate errors that could arise if data is added to a db and a crash occurs before the message could be queued, or vice versa. 
This can be done either by returning a pointer to a tx or by accepting a tx as an input parameter when queuing a message. If this is a feature you could add it would be very appreciated :)

**Comment by craigpastro** [comment]
Hi @nathankawalec! There is `SendBatch`, but perhaps this doesn't work for your use case? If so, then perhaps a `SendTx` makes sense. I'm happy to review a PR, otherwise, I will attempt an implementation when I have some spare time. Thanks!

**Comment by nathankawalec** [comment]
Sorry for the late response. Upon further review, I think all that is required is something like:
`NewFromTx() *PGMQ { ... }`

Only issue I can think of is pgxpool transaction doesn't implement Close or Ping methods, which would require end user to wrap the transaction object in order to implement those methods (with no op)

I believe Close method called on rows in SendBatchWithDelay, ReadBatch, ArchiveBatch, and DeleteBatch can be safely called in a transaction with more work to be done, but I may be mistaken.

I think ideally it would be possible to use each method inside a transaction.

**Comment by craigpastro** [comment]
I finally got around to implementing this in https://github.com/craigpastro/pgmq-go/pull/63. I dropped the `PGMQ` struct and added an argument to all functions that only needs to satisfy the DB interface:
```
type DB interface {
	Exec(ctx context.Context, query string, args ...any) (pgconn.CommandTag, error)
	Query(ctx context.Context, query string, args ...any) (pgx.Rows, error)
	QueryRow(ctx context.Context, query string, args ...any) pgx.Row
}
```
In particular, `pgx.Tx` satisfies this interface.

Example usage:
```
// Suppose pool is an instance of *pgxpool.Pool and a PGQM queue named "queue" exists

tx, err := pool.Begin(ctx)
// handle error

id, err := pgmq.Send(ctx, tx, "queue", json.RawMessage(`{"foo": "bar"}`))
// handle err

// do other transaction stuff
err = tx.Commit(ctx)
```

WDYT?

**Comment by craigpastro** [comment]
Correct. I think it makes sense because you may be doing many other things with the connection. The advantage of using a queue in Postgres. Of course we could provide convenience functions like CreatePgxPool, etc. if those are valuable.

**Comment by craigpastro** [comment]
Hi @softexpert. Oooh, the generics is a nice idea. Thank you for the link. I've added a ["see also" section to the readme](https://github.com/craigpastro/pgmq-go/pull/68) with a link to their package.

**Comment by craigpastro** [comment]
Resolved in #67.
